stellargraph/data/explorer.py

```diff
@@ -780,3 +783,163 @@ def _check_neighbourhood_sizes(self, in_size, out_size):
+            walk_bias (str, optional): distribution to use when choosing a random
+                neighbour to walk through. Available options are:
+                * None (default) - neighbours are picked from a uniform distribution
+                * "exponential" - exponentially decaying probability, resulting in a bias
+                    towards shorter time gaps
+            p_walk_success_threshold (float): Lower bound for the proportion of successful
+                (i.e. longer than minimum length) walks. If the 95% percentile of the
+                estimated proportion is less than the provided threshold, a RuntimeError
+                will be raised.
+            seed (int, optional): Random number generator seed; default is None
+
+        Returns:
+            List of lists of node ids for each of the random walks
+
+        """
+        if cw_size < 2:
+            raise ValueError(
+                f"cw_size: context window size should be greater than 1, found {cw_size}"
+            )
+        if max_walk_length < cw_size:
+            raise ValueError(
+                f"max_walk_length: maximum walk length should not be less than the context window size, found {max_walk_length}"
+            )
+
+        np_rs = self._np_random_state if seed is None else np.random.RandomState(seed)
+        walks = []
+        num_cw_curr = 0
+
+        edges, times = self.graph.edges(include_edge_weight=True)
+        temporal_edges = list(zip(edges, times))
+        edge_biases = self._temporal_biases(
+            times, None, bias_type=initial_edge_bias, is_forward=False,
+        )
+
+        successes = 0
+        failures = 0
+
+        def not_progressing_enough():
+            # (this uses Beta(1, 1) as the prior, since it's uniform on p)
+            posterior = stats.beta(1 + successes, 1 + failures)
+            return posterior.ppf(0.95) < p_walk_success_threshold
+
+        # loop runs until we have enough context windows in total
+        while num_cw_curr < num_cw:
+            (src, dst), t = self._sample(temporal_edges, edge_biases, np_rs)
+
+            remaining_length = num_cw - num_cw_curr + cw_size - 1
+
+            walk = self._walk(
+                src, dst, t, min(max_walk_length, remaining_length), walk_bias, np_rs
+            )
+            if len(walk) >= cw_size:
+                walks.append(walk)
+                num_cw_curr += len(walk) - cw_size + 1
+                successes += 1
+            else:
+                failures += 1
+                if not_progressing_enough():
+                    raise RuntimeError(
+                        f"Discarded {failures} walks out of {failures + successes}. "
+                        "Too many temporal walks are being discarded for being too short. "
+                        "Consider using a smaller context window size."
+                    )
+
+        return walks
+
+    def _sample(self, items, biases, np_rs):
+        chosen_index = (
```

Comment:
I personally find this easier without the parens, i.e.

```python
if biases is not None:
    chosen_index = ...
else:
    chosen_index = ...
```

but totally up to you.